internal/controller/operatorconfigmap_controller.go

```diff
@@ -613,6 +624,21 @@ func (c *OperatorConfigMapReconciler) getDeployCSIConfig() (bool, error) {
 	return deployCSI, nil
 }
 
+func (c *OperatorConfigMapReconciler) getNoobaaSubManagementConfig() (bool, error) {
+	data := c.operatorConfigMap.Data
+	manageNoobaaSub := true
+	if data != nil {
+		var err error
+		if value, ok := data[manageNoobaaSubKey]; ok {
+			manageNoobaaSub, err = strconv.ParseBool(value)
+			if err != nil {
+				return false, fmt.Errorf("failed to parse value for %q in operator configmap as a boolean: %v", manageNoobaaSubKey, err)
+			}
+		}
+	}
+	return manageNoobaaSub, nil
```

Comment:
Failure of a configuration value should not result in a failure of the reconciliation. Assume that any value that cannot be lowercased to "true", as false. 
This entire function could be replaced with a two-liner on the call site. 
```

manageNoobaaSubAsString := cmp.Or(
    c.operatorConfigMap.Data[manageNoobaaSubKey], 
    strconv.FormatBool(true),
) 
manageNooBaaSub: = strings.ToLower(manageNoobaaSubAsString) != strconv.FormatBool(true)
    
```

Reply:
i understand what you mean by that but if we are supposed to manage noobaa and fail to do so because of a typo shouldn't we fail the reconciliation? how would we indicate to the user that something is problematic in his configuration? (the user is not creating this configuration as well it is us who set it up so the typo should not happen in any case)

Reply:
No, reconciliation failure is the worst result because it stops any and all reconciliation processes, related and unrelated to the noobaa subscription.  
When taking a value from an untyped value (like a config map value) consider that any value is deviation from the default behavior. That means that a type from a valid value should be treated as no change from the default behavior. 
As noobaa subscription reconciliation is the default we need to consider that any value that is not specifically false (ignoring case) means we should reconcile noobaa subscription. 

P.S
I have updated the suggestion in the original comment to check for non-equality with `false`